In [1]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
from statsmodels.multivariate.manova import MANOVA
from sklearn.preprocessing import StandardScaler

In [2]:
df=pd.read_csv('../data/06-primas_norm_est_standard.csv',index_col=0)
df.head(10)

age     sex       bmi  children smoker     region  charges_Sklearn
0 -1.387916  female -0.382685 -0.885991    yes  southwest         0.836552
1 -1.459488    male  0.601634 -0.069805     no  southeast         0.032037
2 -0.743769    male  0.472516  1.562566     no  southeast         0.176597
3 -0.457482    male -0.218352 -0.885991     no  northwest         0.145677
4 -0.529054  female -0.744888 -0.885991     no  southeast         0.139827
5  0.544525  female  0.546298 -0.069805     no  southeast         0.377794
6 -0.099622  female -0.409515  1.562566     no  northwest         0.326895
7 -0.099622    male -0.059050  0.746380     no  northeast         0.280453
8 -0.958485    male -0.664398 -0.885991     no  northeast         0.084883
9 -1.101629    male  0.707277 -0.885991     no  southwest         0.037413

# Pair Programming ANOVA

## Ana C y Ana G

En el pair programming de hoy usaremos el set de datos que guardastéis en el pair programming de normalización y estandarización.

Hasta ahora habéis estado evaluando las características de vuestro set de datos y habéis hecho una gran exploración, es el momento de hacer vuestro primer ANOVA! En el ejercicio de hoy tendréis que hacer un ANOVA con vuestro datos y hacer una interpretación de los resultados.

In [4]:
df.head()

age     sex       bmi  children smoker     region  charges_Sklearn
0 -1.387916  female -0.382685 -0.885991    yes  southwest         0.836552
1 -1.459488    male  0.601634 -0.069805     no  southeast         0.032037
2 -0.743769    male  0.472516  1.562566     no  southeast         0.176597
3 -0.457482    male -0.218352 -0.885991     no  northwest         0.145677
4 -0.529054  female -0.744888 -0.885991     no  southeast         0.139827

In [5]:
mi_primera_anova = ols('charges_Sklearn ~ age+sex+bmi+children+smoker+region', data=df).fit()
sm.stats.anova_lm(mi_primera_anova)

df     sum_sq    mean_sq            F         PR(>F)
sex          1.0   0.234488   0.234488    16.566829   5.047142e-05
smoker       1.0  16.590362  16.590362  1172.125783  2.288475e-173
region       3.0   0.201258   0.067086     4.739700   2.732392e-03
age          1.0  33.436069  33.436069  2362.291888  1.654167e-271
bmi          1.0   0.145069   0.145069    10.249264   1.408243e-03
children     1.0   1.016088   1.016088    71.787672   7.965937e-17
Residual  1055.0  14.932555   0.014154          NaN            NaN

Si PR es >=0.05, nuetsras variables predictoras influyen en nuestra variable respuesta, por lo que vemos, que todas nuestras variables, influyen 'charges'.

In [6]:
mi_primera_anova.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:        charges_Sklearn   R-squared:                       0.776
Model:                            OLS   Adj. R-squared:                  0.774
Method:                 Least Squares   F-statistic:                     455.9
Date:                Sun, 29 Jan 2023   Prob (F-statistic):               0.00
Time:                        19:04:42   Log-Likelihood:                 759.89
No. Observations:                1064   AIC:                            -1502.
Df Residuals:                    1055   BIC:                            -1457.
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
=======================================================================================
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept               0.3574      0.008     42.794      0.000       0.341       0.374
sex[T.male]            -0.0245      0.007     -3.361      0.001      -0.039      -0.010
smoker[T.yes]           0.6904      0.016     42.385      0.000       0.658       0.722
region[T.northwest]    -0.0133      0.010     -1.268      0.205      -0.034       0.007
region[T.southeast]    -0.0324      0.011     -3.035      0.002      -0.053      -0.011
region[T.southwest]    -0.0428      0.010     -4.098      0.000      -0.063      -0.022
age                     0.1780      0.004     47.475      0.000       0.171       0.185
bmi                     0.0126      0.004      3.191      0.001       0.005       0.020
children                0.0310      0.004      8.473      0.000       0.024       0.038
==============================================================================
Omnibus:                     1011.235   Durbin-Watson:                   1.970
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            28237.617
Skew:                           4.600   Prob(JB):                         0.00
Kurtosis:                      26.501   Cond. No.                         5.56
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

>- Nos da un R2 de 0.774, es decir, nuestras variables predictoras explican en un 77.4% nuestra variable respuesta.
>- Podemos que ver que todas nuestras variables predictoras excepto sexo (masculino), tienen relación positiva, si aumentan, aumenta el valor de 'charges. 
>- El resto la tienen negativa, a medida que aumentan, disminuye el valor de 'charges'.
>- Con p-value<0.05 tenemos edad, bmi, fumadores, rechazamos H0, nuestras variables predictoras tienen efecto en nuestra variable respuesta.
>- La variable de localización no la estamos tomando en cuenta, desde el inicio del pair no nos salen resultados lógicos.